cpu/atmega_common/periph/i2c.c

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences, Dimitri Nahm
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  driver_periph
+ * @{
+ *
+ * @file
+ * @brief       Low-level I2C driver implementation fot atmega common
+ *
+ * @note This implementation only implements the 7-bit addressing mode.
+ *
+ * @author      Dimitri Nahm <dimitri.nahm@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "cpu.h"
+#include "mutex.h"
+#include "periph/i2c.h"
+#include "periph_conf.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define MT_START 0x08
+#define MT_ADDRESS_ACK 0x18
+#define MT_DATA_ACK 0x28
+#define MR_ADDRESS_ACK 0x40
+
+/* guard file in case no I2C device is defined */
+#if I2C_NUMOF
+
+/* static function definitions */
+static int _start(uint8_t address, uint8_t rw_flag);
+static int _write(const uint8_t *data, int length);
+static void _stop(void);
+
+/**
+ * @brief Array holding one pre-initialized mutex for each I2C device
+ */
+static mutex_t locks[] =  {
+#if I2C_0_EN
+    [I2C_0] = MUTEX_INIT,
+#endif
+};
```

Comment:
as this driver supports only one I2C device anyway, I would prefer to cut this to a single line:
```c
static mutex_t lock = MUTEX_INIT;
```
no need for an array here...